src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -755,6 +757,15 @@ private BoundExpression TransformCompoundAssignmentLHS(BoundExpression originalL
             return variableTemp;
         }
 
+        private BoundExpression VisitCompoundAssignmentReceiver(BoundExpression receiver, bool isRegularCompoundAssignment)
+        {
+            var previousInCompoundAssignmentReceiver = _inCompoundAssignmentReceiver;
+            _inCompoundAssignmentReceiver |= isRegularCompoundAssignment;
+            var loweredReceiver = VisitExpression(receiver);
+            _inCompoundAssignmentReceiver = previousInCompoundAssignmentReceiver;
+            return loweredReceiver;
```

Comment:
It seems like `_inCompoundAssignmentReceiver` will be set for the duration of the `VisitExpression` call. Doesn't that mean that all array accesses within the receiver will end up with the `.readonly` prefix? Essentially I'm thinking about the following: 

```csharp
array1[(array2[0]=someValue).Index].Prop += 42;
```

Will the inner `array2[0]` get the `.readonly` prefix? If so that seems incorrect and would lead to array mismatch bugs.